classes/submissionFile/DAO.inc.php

```diff
@@ -0,0 +1,466 @@
+        }
+
+        if (!isset($submissionFile)) {
+            $submissionFile = $this->get($bestId);
+        }
+
+        if (
+            $submissionFile &&
+            in_array(
+                $submissionFile->getData('fileStage'),
+                [
+                    SubmissionFile::SUBMISSION_FILE_PROOF,
+                    SubmissionFile::SUBMISSION_FILE_DEPENDENT
+                ]
+            )
+        ) {
+            return $submissionFile;
+        }
+
+        return null;
+    }
+
+    /**
+     * Assign file to a review round.
+     *
+     * @param $submissionFileId int The file to be assigned.
+     * @param $reviewRound ReviewRound
+     */
+    public function assignRevisionToReviewRound(
+        int $submissionFileId,
+        ReviewRound $reviewRound
+    ): void {
+        // Avoid duplication errors -- clear out any existing entries
+        $this->deleteReviewRoundAssignment($submissionFileId);
+
+        DB::table('review_round_files')->insert([
+            'submission_id' => (int) $reviewRound->getSubmissionId(),
+            'review_round_id' => (int) $reviewRound->getId(),
+            'stage_id' => (int) $reviewRound->getStageId(),
+            'submission_file_id' => $submissionFileId,
+        ]);
+    }
+
+    /**
+     * Remove a specific file assignment from a review round.
+     */
+    public function deleteReviewRoundAssignment(int $submissionFileId): void
+    {
+        // Remove currently assigned review files.
+        DB::table('review_round_files')->where([
+            'submission_id' => (int) $submissionFileId
+        ])->delete();
+    }
+
+    /**
+     * Checks if public identifier exists (other than for the specified
+     * submission file ID, which is treated as an exception).
+     *
+     * $pubIdType it is one of the NLM pub-id-type values or
+     * 'other::something' if not part of the official NLM list
+     * (see <http://dtd.nlm.nih.gov/publishing/tag-library/n-4zh0.html>).
+     */
+    public function pubIdExists(
+        $pubIdType,
+        $pubId,
+        $excludePubObjectId,
+        $contextId
+    ): bool {
+        $result = DB::table($this->settingsTable . ' as sfs')
+            ->join('submission_files AS sf', 'sfs.submission_file_id', '=', 'sf.submission_file_id')
+            ->join('submissions AS s', 'sf.submission_id', '=', 's.submission_id')
+            ->where([
+                'sfs.setting_name' => 'pub-id::' . (string) $pubIdType,
+                'sfs.setting_value' => (string) $pubId,
+                'sfs.submission_file_id' => (int) $excludePubObjectId,
+                's.context_id' => (int) $contextId
+            ])->count();
+        return (bool) $result > 0;
+    }
+
+    /**
+     * @copydoc PKPPubIdPluginDAO::changePubId()
+     */
+    public function changePubId($pubObjectId, $pubIdType, $pubId)
+    {
+        DB::table($this->settingsTable)
```

Comment:
I think this query is broken; it appears to only change `locale` and possibly `type`, but I think it was intended to change e.g. the setting value as well. See `PKPSubmissionFileDAO::changePubId` and the `$keyColumn` parameter to [ADODB's "replace" function](https://adodb.org/dokuwiki/doku.php?id=v5:reference:connection:replace).

Reply:
Changed it to uses the Builder updateOrInsert method